azurerm/resource_arm_private_link_service_test.go

```diff
@@ -0,0 +1,342 @@
+	standardResources := testAccAzureRMPrivateLinkServiceTemplate_standardResources(rInt, location)
+	privateLink := testAccAzureRMPrivateLinkServiceTemplate_complete(rInt)
+
+	return testAccAzureRMPrivateLinkServiceTemplate(subscriptionDataSource, standardResources, privateLink)
+}
+
+func testAccAzureRMPrivateLinkServiceTemplate(subscriptionDataSourceTemplate string, standardResourcesTemplate string, privateLinkTemplate string) string {
+	return fmt.Sprintf(`
+%s
+
+%s
+
+%s
+`,subscriptionDataSourceTemplate, standardResourcesTemplate, privateLinkTemplate)
+}
+
+func testAccAzureRMPrivateLinkServiceTemplate_subscriptionDataSource() string {
+	return `data "azurerm_subscription" "current" {}`
+}
+
+func testAccAzureRMPrivateLinkServiceTemplate_standardResources(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestvnet-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  address_space       = ["10.5.0.0/16"]
+}
+
+resource "azurerm_subnet" "test" {
+  name                                  = "acctestsnet-%d"
+  resource_group_name                   = azurerm_resource_group.test.name
+  virtual_network_name                  = azurerm_virtual_network.test.name
+  address_prefix                        = "10.5.1.0/24"
+  private_link_service_network_policies = "Disabled"
+}
+
+resource "azurerm_public_ip" "test" {
+  name                = "acctestpip-%d"
+  sku                 = "Standard"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  allocation_method   = "Static"
+}
+
+resource "azurerm_lb" "test" {
+  name                = "acctestlb-%d"
+  sku                 = "Standard"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  frontend_ip_configuration {
+    name                 = azurerm_public_ip.test.name
+    public_ip_address_id = azurerm_public_ip.test.id
+  }
+}
+`,rInt, location, rInt, rInt, rInt, rInt)
+}
+
+func testAccAzureRMPrivateLinkServiceTemplate_basic(rInt int) string {
+	return fmt.Sprintf(`
+resource "azurerm_private_link_service" "test" {
+  name                           = "acctestpls-%d"
+  location                       = azurerm_resource_group.test.location
+  resource_group_name            = azurerm_resource_group.test.name
+
+  nat_ip_configuration {
+    name                         = "primaryIpConfiguration-%d"
+    subnet_id                    = azurerm_subnet.test.id
+  }
+
+  load_balancer_frontend_ip_configuration_ids = [
+    azurerm_lb.test.frontend_ip_configuration.0.id
+  ]
+}
+`, rInt, rInt)
+}
+
+func testAccAzureRMPrivateLinkServiceTemplate_update(rInt int) string {
```

Comment:
Could we just use the standard resources template in here?
```suggestion
func testAccAzureRMPrivateLinkService_update(rInt int) string {
```

Reply:
fixed